src/providers/aws/mod.rs

```diff
@@ -66,6 +78,26 @@ impl AwsProvider {
         format!("{}/{}", URL, key)
     }
 
+    fn fetch_imdsv2_token(client: retry::Client) -> Result<String> {
+        let token: String = client
+            .header(header::HeaderName::from_bytes(b"X-aws-ec2-metadata-token-ttl-seconds")
+                        .chain_err(|| "setting header name for aws imdsv2 token")?,
+                    header::HeaderValue::from_bytes(b"21600")
+                        .chain_err(|| "setting header value for aws imdsv2 token")?)
+            .put(
+                retry::Raw,
+                #[cfg(test)]
```

Comment:
I think you can fold this build-condition back into `endpoint_for()` by adding a new `use_latest: bool` parameter there and passing `true` in the caller here.

Also, small nit, the `XXX: ` here below would better be a `NOTE(zonggen): ` (or whatever nick you prefer to sign yourself).

Reply:
Will proceed with `use_latest:bool` since that sounds very neat :) 
Updating >>